app/src/main/java/com/example/myapplicationwork/fragments/FragmentMoviesList.kt

```diff
@@ -1,39 +1,46 @@
 package com.example.myapplicationwork.fragments
 
+import android.content.Context
 import android.os.Bundle
-import androidx.fragment.app.Fragment
 import android.view.View
-import android.content.Context
-import android.view.LayoutInflater
-import android.view.ViewGroup
+import androidx.fragment.app.Fragment
 import androidx.recyclerview.widget.GridLayoutManager
 import androidx.recyclerview.widget.RecyclerView
-import com.example.myapplicationwork.listeners.ClickListener
 import com.example.myapplicationwork.R
 import com.example.myapplicationwork.adapers.MoviesListAdapter
 import com.example.myapplicationwork.adapers.OnRecyclerItemClicked
+import com.example.myapplicationwork.data.Movie
+import com.example.myapplicationwork.data.loadMovies
 import com.example.myapplicationwork.listeners.ClickListenerOnList
-import com.example.myapplicationwork.modelsClass.Movie
-import com.example.myapplicationwork.modelsClass.MoviesGenerator
 import com.google.android.material.snackbar.Snackbar
-
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
 
 class FragmentMoviesList : Fragment(R.layout.fragment_movies_list) {
-    private var listener: ClickListener? = null
+
     private var listenerList: ClickListenerOnList? = null
     private var recycler: RecyclerView? = null
+    private val scope = CoroutineScope(Dispatchers.Main);
+    private var moviesAdapter: MoviesListAdapter? = null
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        moviesAdapter = MoviesListAdapter(clickListener)
+        scope.launch {
```

Comment:
you don't need to create a scope 
inside fragment has `viewLifecycleOwner.lifecycleScope` coroutine scope which auto-canceled when fragment view destroyed